openhtf/io/output/upload_json.py

```diff
@@ -0,0 +1,72 @@
+"""Module for uploading JSON-formatted test record files to server."""
+
+import os
+import logging
+import requests
+
+from json import JSONEncoder
+from openhtf import conf
+from openhtf.io.output.json_factory import OutputToJSON
+
+conf.Declare('project', 'project name')
+conf.Declare('data_server', 'data server')
+_LOG = logging.getLogger(__name__)
+
+class UploadJSON(OutputToJSON):
+  """Return an output callback that upload JSON Test Records to data server.
+
+  An example filename_pattern might be:
+    '/data/test_records/%(dut_id)s.%(start_time_millis)s'
+
+  To use this output mechanism:
+    test = openhtf.Test(PhaseOne, PhaseTwo)
+    test.AddOutputCallback(UploadJson(
+        '/data/test_records/%(dut_id)s.%(start_time_millis)s'))
+  
+    in config.yaml:
+       data_server:  http://your_data_server_entry
+  Args:
+    filename_pattern: A format string specifying the filename to write to,
+      will be formatted with the Test Record as a dictionary.
+    inline_attachments: Whether attachments should be included inline in the
+      output.  Set to False if you expect to have large binary attachments.
+  """
+
+  def __init__(self, filename_pattern=None, inline_attachments=True, **kwargs):
+    super(UploadJSON, self).__init__(filename_pattern, inline_attachments, **kwargs)
+    self.headers = {'Accept' : 'application/json',
+                 'Content-Type' : 'application/json'}
+    self.data_server = None
+    self.file_dir = filename_pattern.split('%')[0]
+
+  # pylint: disable=invalid-name
+  def __call__(self, test_record):
+    super(UploadJSON, self).__call__(test_record)
+  # pylint: enable=invalid-name
+    if not self.data_server:
+      config = conf.Config()
+      project = config.project
+      server = config.data_server
+   
+      if server:
+        self.data_server = server + project
+      else:
+        raise ValueError('No data_server set')
+    
+    for f in self.IterFile(self.file_dir):
+      resp = requests.post(self.data_server, data=open(f, 'rb'),
+                             headers=self.headers)
+      if resp.status_code == 200:
+        os.remove(f)
+        _LOG.info('Test record successfully loaded to servers.')
+      else:
+        _LOG.warning('Error on upload test_record to server rc:%d, file:%s',
```

Comment:
I think you're going to get a really long log line putting this all on one line, consider splitting it into:
```
_LOG.warning('Error uploading test record to %s: %s - %s', self.upload_target, resp.status_code, resp.reason)
_LOG.warning('Test record saved to %s', record_filename)
```